CMakeLists.txt

```diff
@@ -280,24 +198,27 @@ endif()
 include(CheckCCompilerFlag)
 if(CMAKE_C_COMPILER_ID MATCHES "Clang")
     message(STATUS "libavif: Enabling warnings for Clang")
-    add_compile_options(-Wall -Wextra -Wshorten-64-to-32)
+    target_compile_options(avif PUBLIC $<BUILD_INTERFACE:-Wall -Wextra -Wshorten-64-to-32>)
 elseif(CMAKE_C_COMPILER_ID MATCHES "GNU")
     message(STATUS "libavif: Enabling warnings for GCC")
-    add_compile_options(-Wall -Wextra)
```

Comment:
Since this is in the project root (and not in src/CMakeLists.txt or tests/CMakeLists.txt) it would get applied to third-party libraries added with `add_subdirectory`. You need to wrap these flags in `$<$<OR:$<COMPILE_LANGUAGE:C>,$<COMPILE_LANGUAGE:CXX>>:-flag>` to avoid nasm `-w+unknown-warning` errors. And there is at least one place in SVT-AV1 that fails to compile with `-Werror,-Wshorten-64-to-32`. [Source/Lib/Encoder/C_DEFAULT/variance.c:296](https://gitlab.com/AOMediaCodec/SVT-AV1/-/blob/aeb1867362596903bf88a60f8801219ce7b9ce7b/Source/Lib/Encoder/C_DEFAULT/variance.c#L296):

<blockquote><pre><code>error: implicit conversion loses integer precision: 'long long' to 'uint32_t' (aka 'unsigned int') [-Werror,-Wshorten-64-to-32]
    return *sse - ((int64_t)sad * sad) / (w * h);
    ~~~~~~ ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.</code></pre>

Even if that issue is fixed, I could imagine maintenance issues caused by enforcing more stringent warning/error flags on dependencies than they have for themselves.

Reply:
By making them public compile options they get added to all internal code anyway, because anything that isn't in the avif target links against it.